### ISSUE TITLE
A New Approach to Vehicle Bypass 

### ISSUE THREAD

**Issue by derimmer** [vasl-developers/vasl#936]
We currently have a vehicle bypass counter. I think there are two things that are true about it:

1. Nobody ever remembers where to find it.
2. It adds to the counter clutter and some players find it confusing. 

Given the success of the Bad Things labels on AFVs in reducing counter clutter, I thought a similar approach might work for displaying bypass status (and actually for moving the AFV in to bypass position), as follows:

1. Move the AFV into the hex as if there were no bypass
2. Right-click->Move->Bypass (new item)-> Hexside (new submenu same as moving counter one hex by numpad).
3. VASL rotates the counter and sets it on the hexside as per the Hexside selected
4. VASL adds colour to the corner of the counter closest to the centre of the hex being bypassed. 

This (a) avoids the additional bypass counter and (b) auto-positions the AFV which can sometimes be tricky for non-adept mouse users like me.

Thoughts and suggestions?

I will try to build this into a 6.6.3 beta version and we can get user reaction.

**Comment by hongkongwargamer** [comment]
Question: 

Imagine a counter has already done a bypass and is now sitting on a hexside .. and the user wants to bypass another hexside from there (same hex) .. 

Do I do right-click menu while the counter's still sitting on the first hexside, or do I have to move the counter back to the center of the bypassed hex before doing right-click menu again, just so VASL knows exactly which hex I am bypassing (and therefore color points to the correct hex center)?  Or does it defaults to the last hex until you move the counter to a hex center again?

**Comment by derimmer** [comment]
@hongkongwargamer Good question. I think this only works if you right click again and VASL moves the AFV to the new hexside. If you have to recenter the AFV first and then reselect  the menu, that's too much clicking. 

So, that's what I would try to build.

**Comment by derimmer** [comment]
Somehow I opened two identical issues. Zoltan commented on the other instance. I will close it and copy his comment here.

@zgrose wrote: Corner marker ain't a bad idea. Not a fan of the moving to an edge part since during the MPh that sounds super annoying and creates one system for Infantry and one system for AFV.

@zgrose I am not sure what you mean? Currently when you move an AFV into bypass, don't you position it along the hexside? This would just have VASL do that for you.

**Comment by hongkongwargamer** [comment]
> @hongkongwargamer Good question. I think this only works if you right click again and VASL moves the AFV to the new hexside. If you have to recenter the AFV first and then reselect the menu, that's too much clicking.
> 
> So, that's what I would try to build.

Okay .. there are variations if we "start" from a vehicle that is presently sitting on a hexside.  Lets imagine for this example that we have an AFV on a hexside bypassing the hex on it's right:
* it turns right - continue bypassing the same hex
* it turns right - but bypassing the hex on its (new) left
* it turns left - bypassing the hex on the left
* it turns left - bypassing the hex on its (new) right
* Then there's stopping, starting in reverse and bypassing in reverse (rare)

All this goes away of course if you require that the user first center the counter on the hex he wants to bypass.

**Comment by derimmer** [comment]
> > @hongkongwargamer Good question. I think this only works if you right click again and VASL moves the AFV to the new hexside. If you have to recenter the AFV first and then reselect the menu, that's too much clicking.
> > So, that's what I would try to build.
> 
> Okay .. there are variations if we "start" from a vehicle that is presently sitting on a hexside. Lets imagine for this example that we have an AFV on a hexside bypassing the hex on its right:
> 
> * it turns right - continue bypassing the same hex
> * it turns right - but bypassing the hex on its (new) left
> * it turns left - bypassing the hex on the left
> * it turns left - bypassing the hex on its (new) right
> * Then there's stopping, starting in reverse and bypassing in reverse (rare)
> 
> All this goes away of course if you require that the user first center the counter on the hex he wants to bypass.

I think all of the above can be handled. It comes down to whether or not you want to move from a current bypass position to a new one in (a) the same hex or (b) a new hex. 

If (a) then just use the right-click menu.

If (b) then move the vehicle to that new hex (center position) then use the right click menu - that probably actually helps you opponent see what hex you are bypassing in. 

I think that would work?

**Comment by hongkongwargamer** [comment]
> > > @hongkongwargamer Good question. I think this only works if you right click again and VASL moves the AFV to the new hexside. If you have to recenter the AFV first and then reselect the menu, that's too much clicking.
> > > So, that's what I would try to build.
> > 
> > 
> > Okay .. there are variations if we "start" from a vehicle that is presently sitting on a hexside. Lets imagine for this example that we have an AFV on a hexside bypassing the hex on its right:
> > 
> > * it turns right - continue bypassing the same hex
> > * it turns right - but bypassing the hex on its (new) left
> > * it turns left - bypassing the hex on the left
> > * it turns left - bypassing the hex on its (new) right
> > * Then there's stopping, starting in reverse and bypassing in reverse (rare)
> > 
> > All this goes away of course if you require that the user first center the counter on the hex he wants to bypass.
> 
> I think all of the above can be handled. It comes down to whether or not you want to move from a current bypass position to a new one in (a) the same hex or (b) a new hex.
> 
> If (a) then just use the right-click menu.
> 
> If (b) then move the vehicle to that new hex (center position) then use the right click menu - that probably actually helps you opponent see what hex you are bypassing in.
> 
> I think that would work?

Genius!

**Comment by zgrose** [comment]
> @zgrose I am not sure what you mean? Currently when you move an AFV into bypass, don't you position it along the hexside? This would just have VASL do that for you.

Correct, currently everyone in the world moves their counter onto the hexside for Infantry and AFV when moving into bypass.
If I understand your intent correctly:
- if I'm moving an Infantry into bypass, I would grab that counter and move it onto an hexside. Then I'll do the next MF, etc.
- if I'm moving an AFV into bypass and I would like to activate the "bypass marker", instead of dropping my unit-to-be-in-bypass I have to grab that counter and drop it into the hex center. Then choose a menu option. Then I'll do my next MP, etc.

Putting aside keyboard shortcuts as assistive tools for moving, this is what I think could work "better":

Menu option:
Rotate Bypass Marker  => This would show/spin the bypass marker around the 4 corners.

Now everyone continues to do what they've done for 20 years but they can also access the bypass marker and don't have to deal with the cognitive load of whether they are meant to drop the counter on the hexside or not to do bypass "properly."

In summary, it would be nice to have the ability to show the bypass marker without having extra steps. If I can use the marker w/o having the change the way I move a counter into bypass, I would be happy. If I have to change the way I move into bypass to accommodate a marker that I may not ever need in a scenario, I would be unhappy.

**Comment by derimmer** [comment]
@zgrose thanks for the comments/suggestions Zoltan. Very helpful. 

My thought was that the new option would be in addition to existing practice, it would not replace current ways of moving but would provide another option. 

If I am moving an AFV in bypass and there are no units anywhere near then the current 'drag it on to the hexside' works well, even when spinning and moving in bypass on a second hexside. There is no need to replace/change that. 

Where I was thinking that a new tool would be helpful would be when moving in bypass in hexes with infantry in the buildings/woods. In my experience, that can get messy very fast. It would allow another way of managing bypass but the existing way would continue to work. 

I like your suggestion of being able to trigger a bypass marker from the popup menu rather than dragging it from the counter palette. Whether that bypass marker is an actual counter or a highlight on the vehicle counter is up for testing.

**Comment by derimmer** [comment]
There is a VBM extension that provides a counter to test if VBM is allowed via a given hex-side. Something like it should also be added as part of this work.

**Comment by derimmer** [comment]
So, for 6.6.7, I am going to add new submenu to the vehicular Move menu that will let you (1) add a VBM counter and (2) will then rotate the counter to the correct image depending on the Hexside selection you have made - in this way it will function just like the existing Move One Hex menu and key commands, only without the movement. 

This will save users having to find the counter and rotate it themselves. I have tested it and just need to add it to the Unit58v prototype AFAICS so it's very simple to do.

**Comment by derimmer** [comment]
this will be in 667.
